util/deephash/deephash.go

```diff
@@ -336,3 +344,43 @@ func (h *hasher) hashMapFallback(v reflect.Value) (acyclic bool) {
 	w.WriteString("}\n")
 	return acyclic
 }
+
+func (h *hasher) hashType(t reflect.Type) {
```

Comment:
I wonder whether we oughtn't have a call to hashType at the beginning of print. Maybe that's coming up in a future change. :)

Reply:
I don't think we should call it at the beginning of _every_ `print`, but we might want to consider doing it for the very top-level concrete type.

At present, all `deephash` usages in the code base always pass the same concrete type at the top-level, so it's not a big deal.

Reply:
> we might want to consider doing it for the very top-level concrete type.

...and every interface we encounter, because the concrete types there might vary from call to call?